src/execution/vm/bytecode_generator.cpp

```diff
@@ -500,12 +500,10 @@ void BytecodeGenerator::VisitUnaryOpExpr(ast::UnaryOpExpr *node) {
 void BytecodeGenerator::VisitReturnStmt(ast::ReturnStmt *node) {
   if (node->Ret() != nullptr) {
     LocalVar rv = GetCurrentFunction()->GetReturnValueLocal();
-    LocalVar result;
     if (node->Ret()->GetType()->IsSqlValueType()) {
-      result = VisitExpressionForSQLValue(node->Ret());
-      BuildAssign(rv, result, node->Ret()->GetType());
+      VisitExpressionForSQLValue(node->Ret(), rv);
```

Comment:
@pmenon Could you please check this?

Writing tests is blocked on #1237.

Reply:
Ack, I was wrong. Let's make a copy of the returned value:

```
if (node->Ret()->GetType()->IsSqlValueType()) {
  LocalVar result = VisitExpressionForSQLValue(node->Ret());
  BuildDeref(rv.ValueOf(), result, node->Ret()->GetType());
}
```